### PR TITLE
test: update persistence agent unit tests. Part of #388

### PR DESCRIPTION
**Pull request by evan-hataishi**
**Which issue is resolved by this Pull Request:** 
Related #388

**Description of your changes:**
Update persistence agent unit tests to remove argo related code.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

**Comment by evan-hataishi** [comment]
/hold

Still need to update persistence worker and workflow_saver.

File: backend/src/agent/persistence/worker/metrics_reporter_test.go [hunk 1]
```diff
@@ -109,389 +87,31 @@ func TestReportMetrics_NoMetricsArtifact_NoOP(t *testing.T) {
...
-			NodeId:       "node-1",
-			ArtifactName: "mlpipeline-metrics",
-		},
-		&api.ReadArtifactResponse{
-			Data: []byte("invalid tgz content"),
-		})
+// Removed TestReportMetrics_InvalidMetricsJSON_Fail - specific to argo artifact spec
 
-	err := reporter.ReportMetrics(workflow)
+// Rempoved TestReportMetrics_InvalidMetricsJSON_PartialFail - specific to argo artifact spec
```
Comment (Tomcli) [inline backend/src/agent/persistence/worker/metrics_reporter_test.go:113]:
```suggestion
// Removed TestReportMetrics_InvalidMetricsJSON_PartialFail - specific to argo artifact spec
```

**Review by Tomcli — COMMENTED** [review]
it looks good to me other than the typo. Just listing some TODOs after #395 
1. run `go test -v -cover ./backend/src/agent/...` unit tests in the new travis job

**Pull request merged**